x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/expression/function/scalar/datetime/DateTimeFormatProcessor.java

```diff
@@ -17,55 +18,90 @@
 import java.time.format.DateTimeFormatter;
 import java.time.temporal.TemporalAccessor;
 import java.util.Locale;
+import java.util.Objects;
 
 import static org.elasticsearch.xpack.sql.util.DateUtils.asTimeAtZone;
 
 public class DateTimeFormatProcessor extends BinaryDateTimeProcessor {
 
     public static final String NAME = "dtformat";
+    private static final String[][] JAVA_TIME_FORMAT_REPLACEMENTS = {
+        {"tt", "a"},
+        {"t", "a"},
+        {"dddd", "eeee"},
+        {"ddd", "eee"},
+        {"K", "v"},
+        {"g", "G"},
+        {"f", "S"},
+    };
+    private final Formatter formatter;
 
-    public DateTimeFormatProcessor(Processor source1, Processor source2, ZoneId zoneId) {
+
+    public enum Formatter {
+        FORMAT,
+        DATE_TIME_FORMAT;
+
+        public String getJavaPattern(final Object pattern) {
+            String javaDateFormat = (String) pattern;
+            if (this == FORMAT) {
+
+                for (String[] replacement : JAVA_TIME_FORMAT_REPLACEMENTS) {
+                    if (javaDateFormat.contains(replacement[0])) {
```

Comment:
I think it's not necessary since, if there is no pattern found, no replacement occurs.

Reply:
Removed contains check.